### PR TITLE
Bug 798392 - Reports do not display when Reports JavaScript dependencies are located at a filepath that includes special characters like hash ("#")

### PR DESCRIPTION
**Pull request by christopherlam**
using a guile's uri.scm sanitizing function with expanded charset.

More test cases are available from https://github.com/Sporkmonger/Addressable/blob/master/spec/addressable/uri_spec.rb

### CODE REVIEW

**Comment by christopherlam** [comment]
Ok just checked Windows build, the HTML produced renders well on Firefox but not in gnucash. There's no access to JS console in libwebkit1 unfortunately.

```xml
<script language="javascript" type="text/javascript" src="file://C:/Program%20Files%20%28x86%29/gnucash/share/gnucash/chartjs/Chart.bundle.min.js"></script>
<h3 style='display:none'></h3><div style='width:100.0%;height:100.0%;'>
```

[x.zip](https://github.com/Gnucash/gnucash/files/7776517/x.zip)

**Comment by jralls** [comment]
With the first commit it works for me on Windows in both a GnuCash reports tab and a Chrome window.

The second commit replaces `gnc-path-find-localized-html-file` with `gnc-resolve-file-path`. The latter is more direct than the former and can't be distracted with `$GNC_DOCS_PATH`.

I'm inclined to remove `gnc:html-css-file`: It's not used and the few css files we have are installed in reports directory where `gnc:html-css-file` isn't likely to find them.

File: gnucash/report/html-utilities.scm [hunk 1]
```diff
@@ -413,11 +413,14 @@
...
+;;path must be absolute. On Windows an absolute path begins with a
+;;drive letter followed by a colon.
 (define (make-uri path)
-  (string-append
-   "file://"
-   (uri-encode (gnc:substring-replace path "\\" "/")
+  (let ((uri-path (uri-encode (gnc:substring-replace path "\\" "/")
                #:unescaped-chars unreserved-chars-rfc3986)))
+  (string-append
+   (if (eq? (car (string->list uri-path)) "/") "file://" "file:///")
```
Comment (christopherlam) [inline gnucash/report/html-utilities.scm:422]: `(eq? (car (string->list uri-path)) "/")` to `(char=? (string-ref uri-path 0) #\/)` will avoid creating intermediate list, and we'd want to be careful about equality functions.

https://www.gnu.org/software/guile/manual/html_node/Equality.html

Comment (jralls) [reply gnucash/report/html-utilities.scm]: OK, done.

**Pull request merged**